meson.build

```diff
@@ -211,40 +211,40 @@ if get_option('hsi')
   conf.set('HAVE_HSI', '1')
 endif
 libxmlb = dependency('xmlb', version : '>= 0.1.13', fallback : ['libxmlb', 'libxmlb_dep'])
-if get_option('gusb')
-  gusb = dependency('gusb', version : '>= 0.3.0', fallback : ['gusb', 'gusb_dep'])
+gusb = dependency('gusb', version : '>= 0.3.0', fallback : ['gusb', 'gusb_dep'], required: get_option('gusb'))
+if gusb.found()
   conf.set('HAVE_GUSB', '1')
 endif
-if get_option('sqlite')
-  sqlite = dependency('sqlite3')
+sqlite = dependency('sqlite3', required: get_option('sqlite'))
+if sqlite.found()
   conf.set('HAVE_SQLITE', '1')
 endif
-if get_option('libarchive')
-  libarchive = dependency('libarchive')
-  conf.set('HAVE_LIBARCHIVE', '1')
+libarchive = dependency('libarchive', required: get_option('libarchive'))
+if libarchive.found()
+    conf.set('HAVE_LIBARCHIVE', '1')
 endif
 endif
 libjcat = dependency('jcat', version : '>= 0.1.0', fallback : ['libjcat', 'libjcat_dep'])
 libjsonglib = dependency('json-glib-1.0', version : '>= 1.1.1')
 valgrind = dependency('valgrind', required: false)
-if get_option('curl')
-  libcurl = dependency('libcurl', version : '>= 7.56.0')
+libcurl_daemon = dependency('libcurl', version : '>= 7.56.0', required: get_option('curl'))
+if libcurl_daemon.found()
   conf.set('HAVE_LIBCURL', '1')
-  if libcurl.version().version_compare('>= 7.62.0')
+  if libcurl_daemon.version().version_compare('>= 7.62.0')
     conf.set('HAVE_LIBCURL_7_62_0', '1')
   endif
 endif
-if build_daemon
-  if get_option('polkit')
-    polkit = dependency('polkit-gobject-1', version : '>= 0.103')
-    conf.set('HAVE_POLKIT', '1')
-    if polkit.version().version_compare('>= 0.114')
-      conf.set('HAVE_POLKIT_0_114', '1')
-    endif
-  conf.set_quoted ('POLKIT_ACTIONDIR', polkit.get_pkgconfig_variable('actiondir'))
-  else
-    warning('Polkit is disabled, the daemon will allow ALL client actions')
+polkit = dependency('polkit-gobject-1', version : '>= 0.103', required: get_option('polkit'))
+if polkit.found()
+  conf.set('HAVE_POLKIT', '1')
+  if polkit.version().version_compare('>= 0.114')
+    conf.set('HAVE_POLKIT_0_114', '1')
   endif
+  conf.set_quoted ('POLKIT_ACTIONDIR', polkit.get_pkgconfig_variable('actiondir'))
+else
+  warning('Polkit is disabled, the daemon will allow ALL client actions')
```

Comment:
Note that this warning will now be printed even when the daemon doesn't allow anything, because it is not built.

When running meson with `--fatal-meson-warnings` that means you cannot disable the daemon?

Reply:
Maybe just the warning needs to be the only thing that moves into `build_daemon` block then like
```
if build_daemon and not polkit.found()
    warning('...)
```